doc/feature/restAPI.md

```diff
@@ -0,0 +1,129 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+GET /registry returns the URIs for all registered objects, together with the
+supported REST operation (GET or PUT). Each objet can be queried for its schema
+using a GET operation on the URI of that object, suffixed by /schema.
+
+- Example:
+  - Get all URIs of registered objects:
+    * Request: ``` GET http://hostname:port/registry ```
+    * Response:
+```
+    {
+      { "namespace/foo": ["GET", "PUT"] },
+      { "namespace/bar": ["PUT"] }
+    }
+```
+  - Get JSON schema for a specific object:
+    * Request: ```GET http://hostname:port/namespace/foo/schema```
```

Comment:
In this example namespace looks like a reserved word. Use a_namespace or namespace_x to disambiguate.